src/unix/os390.c

```diff
@@ -603,11 +603,28 @@ int uv_fs_event_stop(uv_fs_event_t* handle) {
     abort();
 
   uv__handle_stop(handle);
+  if (handle->path != NULL) {
+    uv__free(handle->path);
+    handle->path = NULL;
+  }
 
   return 0;
 }
 
 
+static void os390_re_regfileint(uv_fs_event_t* handle) {
+  if (handle == NULL || handle->path == NULL)
+    return;
+  char* savepath = uv__strdup(handle->path);
+  uv__handle_stop(handle);
+  uv__free(handle->path);
+  handle->path = NULL;
+  handle->flags &= ~UV_HANDLE_ACTIVE;
+  uv_fs_event_start(handle, handle->cb, savepath, 0);
+  uv__free(savepath);
```

Comment:
I don't like this code pattern here. Let's move the relevant bits into here, then call this from `uv_fs_event_start`
```suggestion
  uv__os390_epoll* ep;
  _RFIS reg_struct;
  int rc;

  ep = handle->loop->ep;
  assert(ep->msg_queue != -1);
  reg_struct.__rfis_cmd  = _RFIS_REG;
  reg_struct.__rfis_qid  = ep->msg_queue;
  reg_struct.__rfis_type = 1;
  memcpy(reg_struct.__rfis_utok, &handle, sizeof(handle));

  rc = __w_pioctl(path, _IOCC_REGFILEINT, sizeof(reg_struct), &reg_struct);
  if (rc != 0 && errno != EALREADY && errno != ENOENT)
     abort();
```

Secondly, all of the `memcpy` in this file look really weird and suspicious. They all require a size-check STATIC_ASSERT there and a comment at least as to why they are casting memory load/stores to be of the wrong measured size and type. Or some may need to be deleted, or changed to:
```
  memcpy(&reg_struct.__rfis_rftok, &handle->rfis_rftok,
         sizeof(handle->rfis_rftok));
```
or
```
  memcpy(reg_struct.__rfis_rftok, handle->rfis_rftok,
         sizeof(*handle->rfis_rftok));
```
or something else.

Reply:
Ah wait, this is an array, isn't it, so the rules with sizeof and pointer-demotion are weird. Yeah, I see it is `char rfis_rftok[8];` so you first are smashing (non-technical term) a pointer-sized object into there, then copying the whole field later in some cases and loading it with `*(uv_fs_event_t**)` in other cases? That seems valid, but could still use a comment.

Reply:
> I don't like this code pattern here. Let's move the relevant bits into here, then call this from `uv_fs_event_start`

Hi, could you clarify which lines of code you're referring to here please? Are you suggesting to move code from `os390_re_regfileint` into `uv_fs_event_start`? Or are you suggesting to move it the other way around? Or something else?

Reply:
Ok I see. In this case, `os390_re_regfileint` isn't intended to be a helper function. The reason we add this function is because on z/OS [a file is implicitly unregistered when the change notification is sent](https://www.ibm.com/docs/en/zos/2.4.0?topic=descriptions-regfileint). So after we receive the notification in `os390_message_queue_handler`, we need to re-register interest in that same file. 

So to re-register interest, we have to first stop the handle with `uv__handle_stop`, then restart it with `uv_fs_event_start`. The similarities with `uv_fs_event_start` is because `uv_fs_event_start` takes a `filepath` parameter, but we don't need to change the filepath when re-registering. So we extract the `handle->path`, and then pass it right back into `uv_fs_event_start`.

Would it be more clear if we renamed the function to something like `os390_fs_event_restart` instead, and add a comment? We could also adjust `uv_fs_event_start` so that the `filepath` parameter is optional if there's no change.

Reply:
No, the start code can call this function, but this code shouldn't need to call back to `uv_fs_event_start`

Reply:
I made some changes. Please take a look and let me know if this is what you had in mind.